File: C/Assignment6_3.c
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment6_3.c 
// Description : Write a program which accept number from user and find factorial.
// Author      : Aniket Devaba Markad 
// Date        : 16/05/2025
// 
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 

#include<stdio.h>

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 
// 
// Function Name : DisplayFactorial
// Description   : This function is used to display factorial. 
// Input         : int 
// Output        : int
// Author        : Aniket Devaba Markad 
// Date          : 16/05/2025 
// 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 

int long DisplayFactorial(int iNo)
{
   int iCnt = 0;
   int long iFact = 1;

   if(iNo < 0)
   {
        iNo = -iNo;
   }

   for(iCnt = 1;iCnt <= iNo;iCnt++ )
   {
        iFact = iFact * iCnt ;
   }

   return iFact;
}

int main()
{
    int  iValue = 0;
    int long iRet = 0;
    
    printf("Enter number :");
    scanf("%d ",&iValue);

    iRet = DisplayFactorial(iValue);

    printf("Factorial is :%ld\n",iRet);

    return 0;
}
//Time Complexcity is : O(n)

File: Java/Assignment6_3.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment6_3.java
// Description : Accept number from user and find its factorial
// Author      : Aniket Devaba Markad                        
// Date        : 1/1/2026
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Number
{

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name : factorial
// Description   : This function is used to calculate factorial  . 
// Author        : Aniket Devaba Markad 
// Date          : 1/1/2026 
// 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    int factorial(int iNo)
    {
        int ifact = 1;

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        for(int i = 1 ;i <= iNo; i++)
        {
            ifact = ifact * i;
        }
        return ifact;
    }
}

public class Assignment6_3 
{
    public static void main (String a[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("enter the number :");
        int iValue = sobj.nextInt();

        Number nobj = new Number();

       int iRet = nobj.factorial(iValue);

        System.out.println("factorial of number is :" + iRet);


    }
}
